src/units.cpp

```diff
@@ -328,14 +333,26 @@ void Units::addUnit(const std::string &reference, const std::string &prefix, dou
     UnitDefinition ud;
     ud.mReference = reference;
     // Allow all nonzero user-specified prefixes
-    try {
-        int prefixInteger = std::stoi(prefix);
-        if (prefixInteger != 0.0) {
+
+    bool predefinedPrefix = false;
```

Comment:
can't this use the new `convertPrefixToInt()` method?

Reply:
No because we don't know the state of the prefix in this method.  The `convertPrefixToInt()` method is for when the input is well formed.  That is, it is either a known prefix string or a valid CellML integer.